en/book/building_a_simple_module/saving_to_database.md

```diff
@@ -12,12 +19,15 @@ if (isset($_POST['submit'])){
     if (empty($_POST['name'])){
```

Comment:
We should update this to match current best practices:
- avoid direct access to $_POST (of $_GET), use XoopsRequest instead. XoopsRequest abstracts the superglobal access, and provides sanitization of the input.
- addslashes should not be used for escaping strings for database, use $xoopsDB->escape() instead. There are potential character set related issues (some are exploitable) when using the addslashes() function. The escape() method of the database connector is reliable in all cases.

The result could look more like this:
```
if ('POST' === XoopsRequest::getMethod()){
    if (empty(XoopsRequest::getString('name', '')) {
        echo 'Please fill in a name';
    } else {
        //  You should NEVER trust anything that your users post - always first check it.
        // $_POST contains all the information from our form. We're creating a new variable to contain the submitted information.
        // This way, we can easily put them through to the database.
        $name       =   $xoopsDB->escape(XoopsRequest::getString('name'));
        $address    =   $xoopsDB->escape(XoopsRequest::getString('address', ''));
        $tel        =   $xoopsDB->escape(XoopsRequest::getString('tel', ''));
        $email      =   $xoopsDB->escape(XoopsRequest::getString('email', ''));
        // All values are done, let's prepare our SQL-query to store this information
        $query      =   "INSERT INTO ".$xoopsDB->prefix("tutorial_myform")." (name, address, telephone, email) VALUES ('$name', '$address', '$tel', '$email' )";
        // Let's perform the query
        $res        =   $xoopsDB->query($query);
        if (!$res) {
            echo "We've encountered an error: " . $xoopsDB->error;
        } else {
            echo "Data was correctly inserted into the database!";
        }
    }
}
```